src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSourceProvider.cs

```diff
@@ -406,6 +469,13 @@ private SuggestedActionSet OrganizeRefactorings(Workspace workspace, CodeRefacto
                 }
 
                 var selectedSpan = selectedSpans[0];
+
+                // We only support refactorings when selected span intersects with the span that the light bulb is asking for.
+                if (!selectedSpan.IntersectsWith(range))
```

Comment:
shouldn't this check intersection after two snapshotSpan moved to same snapshot?